### PR TITLE
Style edits for Concepts section

### PR DESCRIPTION
**Pull request by theletterf**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/4551

Part of a site-wide sweep for style that I do when I've available cycles. Issues fixed include:

- Typos
- Terminology
- Unnecessary capitalization (see https://developers.google.com/style/capitalization)
- Missing descriptions

### CODE REVIEW

File: content/en/docs/concepts/glossary.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ during program execution. Used by the [`Metric`](#metric)
 ### **API**
 
 Application Programming Interface. In the OpenTelemetry project, used to define
-how telemetry data is generated per [`Data Source`](#data-source).
+how telemetry data is generated per [`Data source`](#data-source).
```
Comment (chalin) [inline content/en/docs/concepts/glossary.md:25]: Unrelated, but links to glossary entries like these shouldn't be in code font. I'll create a separate issue to track this.

**Review by chalin — APPROVED** [review]
Generally LGTM, thanks! See inline for a few comments & suggestions.

File: content/en/docs/concepts/glossary.md (hunk 1 above)
Comment (chalin) [reply content/en/docs/concepts/glossary.md:25]: Issue:

- #4551

Comment (theletterf) [reply content/en/docs/concepts/glossary.md:25]: Agreed! Let me clean that up.

Comment (theletterf) [reply content/en/docs/concepts/glossary.md]: Done!

**Review by chalin — APPROVED** [review]
Great, thanks for the extra glossary CSS style cleanup!

**Pull request merged**